src/dmd/expressionsem.d

```diff
@@ -5981,12 +5981,9 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             ob.writestring(")");
             ob.writenl();
         }
-        if (global.params.makeDeps && global.params.oneobj)
+        if (global.params.emitMakeDeps)
         {
-            OutBuffer* ob = global.params.makeDeps;
-            ob.writestringln(" \\");
-            ob.writestring("  ");
-            escapePath(ob, toPosixPath(name));
+            global.params.makeDeps.push(name);
```

Comment:
This one would be a bit trickier to do, but still doable, without using `toDString`, since many (most?) `Filename` functions have a `const(char)[]` overload, including `FileName.searchPath`, from where the `name` variable is derived.